controller/controller_server/csi_controller_server.py

```diff
@@ -485,46 +394,25 @@ def DeleteSnapshot(self, request, context):
                 logger.debug(array_mediator)
                 try:
                     array_mediator.delete_snapshot(snapshot_id)
-
                 except array_errors.ObjectNotFoundError as ex:
                     logger.debug("Snapshot was not found during deletion: {0}".format(ex))
 
         except array_errors.ObjectNotFoundError as ex:
             logger.debug("snapshot was not found during deletion: {0}".format(ex.message))
             context.set_code(grpc.StatusCode.OK)
             return csi_pb2.DeleteSnapshotResponse()
-        except array_errors.ObjectIsStillInUseError as ex:
-            logger.info("could not delete snapshot while in use: {0}".format(ex.message))
-            context.set_code(grpc.StatusCode.FAILED_PRECONDITION)
-            context.set_details(ex.message)
-            return csi_pb2.DeleteSnapshotResponse()
-        except array_errors.PermissionDeniedError as ex:
-            context.set_code(grpc.StatusCode.PERMISSION_DENIED)
-            context.set_details(ex.message)
-            return csi_pb2.DeleteSnapshotResponse()
-        except (ValidationException, array_errors.IllegalObjectID) as ex:
-            logger.exception(ex)
-            context.set_details(ex.message)
-            context.set_code(grpc.StatusCode.INVALID_ARGUMENT)
-            return csi_pb2.DeleteSnapshotResponse()
-        except Exception as ex:
-            logger.debug("an internal exception occurred")
-            logger.exception(ex)
-            context.set_code(grpc.StatusCode.INTERNAL)
-            context.set_details('an internal exception occurred : {}'.format(ex))
-            return csi_pb2.DeleteSnapshotResponse()
 
         logger.debug("generating delete snapshot response")
         res = csi_pb2.DeleteSnapshotResponse()
         logger.info("finished DeleteSnapshot")
         return res
 
+    @handle_common_exceptions(csi_pb2.GetCapacityResponse)
     def GetCapacity(self, request, context):
         logger.info("GetCapacity")
-        context.set_code(grpc.StatusCode.UNIMPLEMENTED)
-        logger.info("finished GetCapacity")
-        return csi_pb2.GetCapacityResponse()
+        raise NotImplementedError()
 
+    @handle_common_exceptions(csi_pb2.ControllerExpandVolumeResponse)
```

Comment:
so could we move the `validate_expand_volume_request` call to be outside the `try`?

Reply:
Done